File: vvd-extractor/src/main/java/mikufan/cx/vvd/extractor/ExtractorApplication.java
package mikufan.cx.vvd.extractor;

import mikufan.cx.vvd.extractor.service.MainService;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.context.properties.ConfigurationPropertiesScan;

/**
 * @author CX无敌
 * @date 2020-12-28
 */
@SpringBootApplication
@ConfigurationPropertiesScan
public class ExtractorApplication {
  public static void main(String[] args) {
    var applicationContext = SpringApplication.run(ExtractorApplication.class, args);
    applicationContext.getBean(MainService.class).run();
    System.exit(SpringApplication.exit(applicationContext));
  }
}

File: vvd-extractor/src/main/kotlin/mikufan/cx/vvd/extractor/ExtractorApplication.kt
package mikufan.cx.vvd.extractor

import mikufan.cx.vvd.extractor.service.MainService
import org.springframework.beans.factory.getBean
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.ConfigurationPropertiesScan
import org.springframework.boot.runApplication

/**
 * @date 2022-06-11
 * @author CX无敌
 */
@SpringBootApplication
@ConfigurationPropertiesScan
class ExtractorApplication

fun main(args: Array<String>) {
  runApplication<ExtractorApplication>(*args).use { it.getBean<MainService>().run() }
}
